src/web/routes/application/middleware/handle-path-request.js

```diff
@@ -0,0 +1,27 @@
+const { equals } = require('ramda')
+const { CHECK_URL } = require('../common/constants')
+
+const getPathsInSequence = (steps) => [...steps.map(step => step.path), CHECK_URL]
+
+const isPathAllowed = (sequence, allowed, path) =>
+  sequence.findIndex(equals(path)) <= sequence.findIndex(equals(allowed))
+
+const middleware = (pathsInSequence) => (req, res, next) => {
+  if (!req.session.nextAllowedStep) {
+    req.session.nextAllowedStep = pathsInSequence[0]
+  }
+
+  if (!isPathAllowed(pathsInSequence, req.session.nextAllowedStep, req.path)) {
+    return res.redirect(req.session.nextAllowedStep)
+  }
+
+  next()
+}
+
+const handlePathRequest = (steps) => middleware(getPathsInSequence(steps))
```

Comment:
I have discussed with @YetAnotherMatt maybe changing the name of this to `handleRequestForPath` to stop any confusion that this is related to HTTP `request`. @mikeSmith08 how do you feel about that?

Reply:
Yes, I'm happy with that.